ESPWebThingAdapter.h

```diff
@@ -167,6 +167,33 @@ class WebThingAdapter {
         prop["type"] = "string";
         break;
       }
+
+      prop["readOnly"] = property->readOnly;      
+      
+      if (property->unit != "")
+      {
+        prop["unit"] = property->unit;
+      }
+
+      const char **enumVal = property->propertyEnum;
+      bool hasEnum = false;
+
+      while (property->propertyEnum != nullptr && (*enumVal) != nullptr && !hasEnum)
```

Comment:
This declaration and while loop could be simplified as `bool hasEnum = (property->propertyEnum != nullptr) && ((*property->propertyEnum) != nullPtr);`

Reply:
applied